### PR TITLE
[#137] decrease legion build time

### LINKED ISSUE #137: Decrease Legion build time
Legion build process takes about 25 minutes for now. We should think about increasing speed of the process to improve our CI pipeline. One of the possible ways is to execute some steps in parallel, for example docker images build, tests, etc.

### PR DESCRIPTION
**Pull request by alexander-semenets**
resolves #137 

Build time decreased from about 6mins to 2mins

### CODE REVIEW

**Comment by alexander-semenets** [comment]
ok to test

File: Jenkinsfile [hunk 1]
```diff
@@ -6,204 +6,237 @@ class Globals {
...
+                    cd k8s/grafana
+                    docker build $dockerCacheArg --build-arg pip_extra_index_params=" --extra-index-url ${params.PyPiRepository}" --build-arg pip_legion_version_string="==${Globals.baseVersion}+${Globals.localVersion}" -t legion/k8s-grafana .
+                    """
+                    sh """
+                    docker tag legion/k8s-grafana ${params.DockerRegistry}/legion/k8s-grafana:latest
+                    docker tag legion/k8s-grafana ${params.DockerRegistry}/legion/k8s-grafana:${Globals.baseVersion}
+                    docker tag legion/k8s-grafana ${params.DockerRegistry}/legion/k8s-grafana:${Globals.baseVersion}-${Globals.localVersion}
+                    docker push ${params.DockerRegistry}/legion/k8s-grafana:latest
+                    docker push ${params.DockerRegistry}/legion/k8s-grafana:${Globals.baseVersion}
+                    docker push ${params.DockerRegistry}/legion/k8s-grafana:${Globals.baseVersion}-${Globals.localVersion}
```
Comment (kirillmakhonin) [inline Jenkinsfile]: I think it can be moved to function (a lot of repetitive code)

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
skip tests

**Comment by alexander-semenets** [comment]
@akharlamov the CI tests passed "Build status: SUCCESS"^
Could you please approve the changes if it's ok to you.

**Pull request merged**